services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtDataPublisher.java

```diff
@@ -81,6 +85,7 @@ public void notifyError(Exception exception) {
     }
 
     public void deliverData(ByteBuffer byteBuffer) {
+        log.trace(() -> "received data of size: " + byteBuffer.remaining());
         // If the subscription is cancelled, no op
         if (isDone) {
             return;
```

Comment:
Will it be helpful to log this scenario when deliverData  is called on a cancelled subscription ?

Reply:
It'll probably generate a lot of logs because CRT does not fully support cancelling and deliverData will be invoked continuously until the download is finished.